src/ui/components/Explorer.js

```diff
@@ -49,44 +71,108 @@ export class Explorer extends React.Component<ExplorerProps, ExplorerState> {
       params: {...view.params()},
       showWeightConfig: false,
       recalculating: false,
+      anchorEl: null,
+      selectedIndex: 1,
+      name: null,
     };
   }
 
+  handleMenuClose = () => {
+    this.setState({
+      anchorEl: null,
+    });
+  };
+
   // Renders the dropdown that lets the user select a type
   renderFilterSelect() {
+    const plugins = this.state.view.plugins();
     const optionGroup = (declaration: PluginDeclaration) => {
       const header = (
-        <option
+        <MenuItem
           key={declaration.nodePrefix}
           value={declaration.nodePrefix}
           style={{fontWeight: "bold"}}
+          onClick={() =>
+            this.setState({
+              selectedIndex: 0,
+              anchorEl: null,
+              filter: declaration.nodePrefix,
+              name: declaration.name,
+            })
+          }
         >
           {declaration.name}
-        </option>
+        </MenuItem>
       );
-      const entries = declaration.nodeTypes.map((type) => (
-        <option key={type.prefix} value={type.prefix}>
+      const entries = declaration.nodeTypes.map((type, index) => (
+        <MenuItem
+          key={type.prefix}
+          value={type.prefix}
+          onClick={() =>
+            this.setState({
+              selectedIndex: index + 1,
+              anchorEl: null,
+              filter: type.prefix,
+              name: type.name,
+            })
+          }
+        >
           {"\u2003" + type.name}
-        </option>
+        </MenuItem>
       ));
       return [header, ...entries];
     };
     return (
-      <label>
-        <span style={{marginLeft: "5px"}}>Showing: </span>
-        <select
-          value={NullUtil.orElse(this.state.filter, "")}
-          onChange={(e) => {
-            const filter = e.target.value || null;
-            this.setState({filter});
-          }}
+      <>
+        <List component="div" aria-label="Device settings">
+          <ListItem
+            button
+            aria-haspopup="true"
+            aria-controls="filter-menu"
+            aria-label="filters"
+            onClick={(event) =>
+              this.setState({
+                anchorEl: event.currentTarget,
+              })
+            }
+          >
+            <ListItemText
+              primary={NullUtil.orElse(this.state.name, "Filter")}
+            />
+            {this.state.anchorEl ? (
+              <KeyboardArrowUpIcon />
+            ) : (
+              <KeyboardArrowDownIcon />
+            )}
+          </ListItem>
+          <Divider style={{backgroundColor: "#F20057", height: "2px"}} />
+        </List>
+
+        <Menu
+          id="lock-menu"
+          anchorEl={this.state.anchorEl}
```

Comment:
Let's move the dropdown to below the the actual toggle:
```suggestion
          anchorEl={this.state.anchorEl}
          getContentAnchorEl={null}
          anchorOrigin={{vertical: "bottom"}}
          transformOrigin={{vertical: "top"}}
```